File: src/main/java/com/wawey/processing/spawner/Spawner.java
package com.wawey.processing.spawner;

/**
 * @author Tomas Perez Molina
 */
public interface Spawner<T> extends Observable<SpawnObserver<T>> {
    void addObserver(SpawnObserver<T> observer);
    void removeObserver(SpawnObserver<T> observer);
}

File: src/main/kotlin/com/wawey/processing/model/spawner/Spawner.kt
package com.wawey.processing.model.spawner

import com.wawey.processing.model.entity.Observable

/**
 *
 * @author Tomas Perez Molina
 */
interface Spawner<T>: Observable<SpawnObserver<T>> {
    fun spawn(x: Int, y: Int): T
}

interface SpawnObserver<T> {
    fun notifySpawn(t: T)
}